consul/consul
# User recomendations

What
====
Show recommended proposals and debates to logged in users based in their interests.

How
===
**Admin**
Allow recommended module activation/deactivation on global settings form admin panel. Added new Setting key “feature.user.recommendations” to achieve this. Maybe some forks do not need this feature.

**FrontEnd**
1. Display user recommendations (Debates and Proposals) on Home page only when user is logged
  1.a Use user interests if any (tags from followed objects) to get best user recommentations sorted by votes count
  1.b Discard already followed object from recommendations
2. Add link to home recommendations to go to filtered index
3. Add new settings "feature.user.recommendations" 
4. Add new “Recommended” search filter to Proposals and Debates index pages.

Screenshots
===========
- Partial recommendeds and link (1, 2)
![Home with recommendations](https://user-images.githubusercontent.com/16189/28967448-7662adce-791b-11e7-9313-1175c2616412.png)

- Admin Panel (3)
![Global settings with feature recommendations](https://user-images.githubusercontent.com/16189/28967474-912e0af4-791b-11e7-9d35-bc0e846a8acd.png)

- Recommendations filter (4)
![New filter on Debate index](https://user-images.githubusercontent.com/16189/28967840-64fd1fea-791d-11e7-81dd-a084c4541966.png)

Test
====
- Check display recommendations on Home page (by user logged, admin settings and results)
- Check order by votes on recommendations methods (Proposal and Debate)
- Check recommendations results are related with user interests
- Check recommendation filter is compatible with search by terms

Deployment
==========
Remember we added next settings:
- seeds: `Setting['feature.user.recommendations'] = nil`
- dev_seeds: `Setting.create(key: 'feature.user.recommendations', value: "true")`
